packages/next/client/image.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react'
+
+const loaders: { [key: string]: (props: LoaderProps) => string } = {
+  imgix: imgixLoader,
+  cloudinary: cloudinaryLoader,
+  default: defaultLoader,
+}
+type ImageData = {
+  hosts: {
+    [key: string]: {
+      path: string
+      loader: string
+    }
+  }
+  breakpoints?: number[]
+}
+type ImageProps = {
+  src: string
+  host: string
+  sizes: string
+  breakpoints: number[]
+  unoptimized: boolean
+  rest: any[]
+}
+
+let imageData: ImageData
+if (typeof window === 'undefined') {
+  // Rendering on a server, get image data from env
+  imageData = JSON.parse(process.env.__NEXT_IMAGE_OPTS || '')
```

Comment:
`process.env.__NEXT_IMAGE_OPTS` can just be replaced by webpack like other values, this would avoid a lot of the custom passing. E.g. the below lines where it reads from window would not be needed